### DISCUSSION TITLE
PorousFlow - Heat injection in wells

### DISCUSSION THREAD

**Discussion by burritotukituki** [idaholab/moose#25500] [Q&A Modules: Porous Flow]
Hi all,
I have a TH system in which water is injected and recovered through various wells at different locations within the model. In my current simulation setup, I am utilizing the `PorousFlowPolyLineSink` method to control fluid flux rather. Specifically, I have employed `PorousFlowPolyLineSink` for managing pore pressure, while I have imposed fixed Dirichlet boundary conditions for temperature. I am seeking your advice on whether this is the most appropriate approach.

However, I have encountered an issue when attempting to implement PorousFlowPolyLineSink in conjunction with temperature. It appears that my simulation diverges in this configuration, as exemplified below:
```
  [DiracKernels]  
  [heat_in]
        type = PorousFlowPolyLineSink
        SumQuantityUO = source_hot_heat
        fluxes = -1
        p_or_t_vals = 0.0
        line_length = 10.0
        multiplying_var = temp
        use_enthalpy = true
        point_file = inj.bh
        variable = temperature
    []
[]

  [AuxVariables]
    [temp]
      initial_condition = 293
    []
[]
```
Any suggestions of how can I set this polyline?

**Comment by cpgr** [comment]
Is it just because you have a zero value for `p_or_t_vals` so it is trying to inject 0 heat?

**Reply by burritotukituki** [reply]
@cpgr 

It doesnt matter what value I set there, still injects cold fluid...
I am bit confused, in this example (modules/combined/examples/geochem-porous_flow/geotes_2D/porous_flow.i) every 'p_or_t_vals' is set to zero.
Can you please check this minimal example and let me know what am I doing wrong? Many thanks :)
https://github.com/burritotukituki/PorousFlowStuff/blob/main/min_chris.i

Thanks!!

**Reply by cpgr** [reply]
OK, that's not it then! Can you post your `inj.bh` file too?

**Reply by burritotukituki** [reply]
Sure, many thanks!!
The inj.bh is here
https://github.com/burritotukituki/PorousFlowStuff/blob/main/inj.bh

**Reply by burritotukituki** [reply]
Hi @cpgr 
I write just to reaming you about this issue :)
